### DISCUSSION TITLE
Combined application built from standalone applications?

### DISCUSSION THREAD

**Discussion by viveleroi** [vercel/turborepo#7250] [Help] [answered]
We're evaluating monorepo tools like turbo for our project. 

We have several "applications" that all use a single UI framework and server architecture. However, our deployment can be complex because:

1. An application can be deployed by itself
2. Any number of applications can be deployed together, functioning as if they were a single app
3. All applications can be deployed, as if they were one app

For example say we have applications `cats`, `dogs`, and `birds`. A client pays us for the `cats` application so that's all they get.

Another client wants `cats` and `dogs` so they get an application that when it runs, they have navigation links in the header `Cats | Dogs` and navigating between them feels like navigating any website because they coexist.

A third client wants everything so their app looks like `Cats | Dogs | Birds`. 

Is this possible within turborepo?

I know they can be run individually as separate applications, but we need the ability to combine them.

**Comment by anthonyshew** [comment, marked as answer]
What you're describing here is more of an infrastructural, architectural concern that seems to generally be out of Turborepo's scope to me.

What you're describing is what we refer to as a "multi-zone" application. You can find an example here:
- Template page: https://vercel.com/templates/next.js/microfrontends
- Demo: https://github.com/vercel/examples/tree/main/solutions/microfrontends
- Code: https://github.com/vercel/examples/tree/main/solutions/microfrontends

As far as selecting which of the three apps you want to deploy, the strategy that I would employ would be:
- Setting up environment variables to show/hide content and links for the various sites (for examples, `BIRDS_ENABLED` set to 1 would show the links to the Birds pages)
- Setting up whichever projects you want to be a part of the deployed application (Don't want the Birds microfrontend? Don't deploy it!)

Hope that helps!
